## Урок 2. Дискретные распределения вероятностей

Условие:

1. Вероятность того, что стрелок попадет в мишень, выстрелив один раз, равна 0.8. Стрелок выстрелил 100 раз. Найдите вероятность того, что стрелок попадет в цель ровно 85 раз.

Испольлуем формулу Бернулли.  
n = 100  
k = 85  
p = 0.8  
q = 1 - 0.8 = 0.2  
P(X = k) = (100! / (85! *(100 - 85)!)) * 0.8^85 * 0.2^15

In [6]:
from math import factorial, exp
def bernulli(n, k, p):
    comb=factorial(n)/(factorial(k)*factorial(n-k))
    return comb*(p**k)*(1-p)**(n-k)

In [7]:
print(f'P = {bernulli(100,85,0.8): .2f}')

P =  0.05


Ответ: P = 0.05

2. Вероятность того, что лампочка перегорит в течение первого дня эксплуатации, равна 0.0004. В жилом комплексе после ремонта в один день включили 5000 новых лампочек. Какова вероятность, что ни одна из них не перегорит в первый день? Какова вероятность, что перегорят ровно две?

Используем формулу Пуассона  
Pm = (λ^m/m!)*e^−λ  
λ = p * n  
m = 0  
n = 5000  
p = 0.0004

In [11]:
def poisson_distr(m,p,n):
    lamb=p*n
    return exp(-lamb)*(lamb**m)/factorial(m)

In [12]:
print(f'P = {poisson_distr(0,0.0004,5000): .2f}')

P =  0.14


Ответ: вероятность, что ни одна из лампочек не перегорит в первый день P = 0.14

По второму условию m = 2

In [14]:
print(f'P = {poisson_distr(2,0.0004,5000): .2f}')

P =  0.27


Ответ: вероятность, что перегорят ровно две лампочки P = 0.27

3. Монету подбросили 144 раза. Какова вероятность, что орел выпадет ровно 70 раз?

Используем формулу Бернулли  
n = 144  
k = 70  
p = 0.5

In [16]:
print(f'P = {bernulli(144,70,0.5): .2f}')

P =  0.06


Ответ: P = 0.06

4. В первом ящике находится 10 мячей, из которых 7 - белые. Во втором ящике - 11 мячей, из которых 9 белых. Из каждого ящика вытаскивают случайным образом по два мяча. Какова вероятность того, что все мячи белые? Какова вероятность того, что ровно два мяча белые? Какова вероятность того, что хотя бы один мяч белый?

Решим с помощью условной вероятности  
P(A + B) = (7 / 10)*(6 / 9)*(9 / 11)*(8 / 10)

In [19]:
P1 = (7 / 10) * (6 / 9) * (9 / 11) * (8 / 10)
P1

0.3054545454545454

Ответ: вероятность того, что все мячи белые P1 = 0.3

In [20]:
P2 = ((7/10)*(6/9)*(2/11)*(1/10))+4*((7/10)*(3/9)*(9/11)*(2/10))+((3/10)*(2/9)*(9/11)*(8/10))
P2

0.20484848484848484

Ответ: вероятность того, что ровно два мяча белые P2 = 0.2

In [24]:
P3 = 1 - (3 / 10) * (2 / 9) * (2 / 11) * (1 / 10)
P3

0.9987878787878788

Ответ: вероятность того, что хотя бы один мяч белый P3 = 0.9988